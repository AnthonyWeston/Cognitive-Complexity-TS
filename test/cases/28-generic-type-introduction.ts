function declaration<T extends true & false>() {}

const expression = function<T extends true & false>() {};

const arrow = <T extends true & false>() => {};

class ClassMethod {
    method<T extends true & false>() {}
}

const objectMethod = <T extends true & false>() => {};
