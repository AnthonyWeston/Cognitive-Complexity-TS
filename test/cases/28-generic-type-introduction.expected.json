{
    "28-generic-type-introduction.ts": {
        "score": 5,
        "inner": [
            {
                "name": "declaration",
                "score": 1,
                "line": 1,
                "column": 1,
                "inner": [
                    {
                        "name": "T",
                        "score": 1,
                        "line": 1,
                        "column": 1
                    }

                ]
            },
            {
                "name": "expression",
                "score": 1,
                "line": 3,
                "column": 20,
                "inner": [
                    {
                        "name": "T",
                        "score": 1,
                        "line": 3,
                        "column": 20
                    }
                ]
            },
            {
                "name": "arrow",
                "score": 1,
                "line": 5,
                "column": 15,
                "inner": [
                    {
                        "name": "T",
                        "score": 1,
                        "line": 5,
                        "column": 12
                    }
                ]
            },
            {
                "name": "ClassMethod",
                "score": 1,
                "line": 7,
                "column": 1,
                "inner": [
                    {
                        "name": "method",
                        "score": 1,
                        "line": 8,
                        "column": 5,
                        "inner": [
                            {
                                "name": "T",
                                "score": 1,
                                "line": 8,
                                "column": 12
                            }
                        ]
                    }
                ]
            },
            {
                "name": "objectMethod",
                "score": 1,
                "line": 11,
                "column": 22,
                "inner": [
                    {
                        "name": "T",
                        "score": 1,
                        "line": 11,
                        "column": 23
                    }
                ]
            }
        ]
    }
}